conda/cli/conda_argparse.py

```diff
@@ -221,13 +223,26 @@ def __init__(self, *args, add_help=True, **kwargs):
         if add_help:
             add_parser_help(self)
 
+    @contextmanager
+    def _reraise_argument_error_sorted(self, action, value):
+        try:
+            yield
+        except ArgumentError:
+            # reraise with sorted choices
+            choices = ", ".join(sorted(map(repr, action.choices)))
+            raise ArgumentError(
+                action, f"invalid choice: {value!r} (choose from {choices})"
+            )
+
     def _check_value(self, action, value):
         # extend to properly handle when we accept multiple choices and the default is a list
         if action.choices is not None and isiterable(value):
             for element in value:
-                super()._check_value(action, element)
+                with self._reraise_argument_error_sorted(action, element):
+                    super()._check_value(action, element)
         else:
-            super()._check_value(action, value)
+            with self._reraise_argument_error_sorted(action, value):
+                super()._check_value(action, value)
```

Comment:
Actually, let's not catch the argument error since it removes the ability to translate the error (those `_()` calls in https://github.com/python/cpython/blob/fb0b642bf1aa3ec276304b7170deedd5040c1698/Lib/argparse.py#L2501-L2507) and we lose the ability to benefit from upstream features like the recently implemented automatic suggestion for typos that will be part of Python 3.14 (https://github.com/python/cpython/blob/6d3b5206cfaf5a85c128b671b1d9527ed553c930/Lib/argparse.py#L2575-L2598).

Let's use this instead which only works for our already implemented greedy parser and simply sorts the choices directly:

```python
def _check_value(self, action, value):
    # For our greedy subparsers, sort the choices by their repr for stable output
    if isinstance(action, _GreedySubParsersAction) and isinstance(
        action.choices, dict
    ):
        action.choices = dict(sorted(action.choices.items()))
    # extend to properly handle when we accept multiple choices and the default is a list
    if action.choices is not None and isiterable(value):
        for element in value:
            super()._check_value(action, element)
    else:
        super()._check_value(action, value)
```

No need for the context manager and catching of the argument error anymore.